tiingo/api.py

```diff
@@ -128,6 +136,36 @@ def get_ticker_metadata(self, ticker, fmt='json'):
         elif fmt == 'object':
             return dict_to_object(data, "Ticker")
 
+    def _invalid_frequency(self, frequency):
+        """
+        Check to see that frequency was specified correctly
+        :param frequency (string): frequency string
+        :return (boolean):
+        """
+        daily_freqs = ['daily', 'weekly', 'monthly', 'annually']
+
+        if frequency.lower() in daily_freqs or re.match(self._frequency_pattern, frequency):
+            return False
+        else:
+            return True
+
+    def _get_url(self, frequency):
+        """
+        Return url based on frequency.  Daily, weekly, or yearly use Tiingo
+        EOD api; anything less than daily uses the iex intraday api.
+        :param frequency (string): valid frequency per Tiingo api
+        :return (string): url
+        """
+        if self._invalid_frequency(frequency):
+            etext = ("Error: {} is an invalid frequency.  Check Tiingo API documentation "
+                     "for valid EOD or intraday frequency format.")
+            raise InvalidFrequencyError(etext.format(frequency))
+        else:
+            if frequency.lower() in ['daily', 'weekly', 'monthly', 'annually']:
```

Comment:
Just in case this list of intraday frequency every changes, we'd like to make the code change once in both places (here and in the frequency validation function). We can do

```python
if self._is_intraday_frequency(frequency):
    return "iex/{}/prices"
else:
   return "tiingo/daily/{}/prices" 

def _is_intraday_frequency(frequency):
    return frequency.lower() in ['daily', 'weekly', 'monthly', 'annually']
```

Reply:
That's good too - changed.